package-res/reportviewer/reportviewer.js

```diff
@@ -826,12 +830,36 @@ define([ 'common-ui/util/util', 'common-ui/util/timeutil', 'common-ui/util/forma
           dlg.showBackgroundBtn(_Messages.getString('FeedbackScreenBackground'));
         }else {
           scheduleScreenBtnCallbacks.shift();
+          dlg.hideBackgroundBtn();
         }
 
         dlg.callbacks = scheduleScreenBtnCallbacks;
         return dlg;
       },
 
+      renderBackgroundBtnListener: function (editMode) {
+        var dlg2 = registry.byId('feedbackScreen');
+        var scheduleScreenBtnCallbacks = _scheduleScreenBtnCallbacks;
+        if (dlg2) {
+          if (editMode) {
+            dlg2.hide();
+            scheduleScreenBtnCallbacks.shift();
+            dlg2.hideBackgroundBtn();
```

Comment:
is it safe to modify this array in the "shift" operation here? What happens if you toggle between edit mode and view mode multiple times? 

Wouldnt it be safer to treat the array as immutable and to have 

    dlg2.callbacks = [ scheduleSchreenBtnCallbacks[1]]; 

here and 

     dlg2.callbacks = scheduleScreenBtnCallbacks; 

in the else branch?